gax/pagination.h

```diff
@@ -51,25 +74,185 @@ class PageResult {
     }
     bool operator!=(iterator const& rhs) const { return !(*this == rhs); }
 
+   private:
+    friend PageResult;
     iterator(FieldIterator current) : current_(std::move(current)) {}
 
     FieldIterator current_;
   };
-  // Note: Since the constructor will eventually be private (with friend
-  // access), we don't have to define all the variants for references.
-  // We just want to take ownership of a page.
-  PageResult(PageType&& raw_page, ElementAccessor accessor)
-      : raw_page_(std::move(raw_page)), accessor_(accessor) {}
 
-  iterator begin() { return accessor_(raw_page_)->begin(); }
-  iterator end() { return accessor_(raw_page_)->end(); }
+  PageResult(PageType const& raw_page) : raw_page_(raw_page) {}
+  PageResult(PageType&& raw_page) : raw_page_(std::move(raw_page)) {}
 
+  iterator begin() { return ElementAccessor{}(raw_page_)->begin(); }
+  iterator end() { return ElementAccessor{}(raw_page_)->end(); }
+
+  // Const overloads
+  iterator begin() const { return ElementAccessor{}(raw_page_)->cbegin(); }
+  iterator end() const { return ElementAccessor{}(raw_page_)->cend(); }
+
+  /**
+   * @brief Get the next_page_token for the page.
+   *
+   * @retun the token for the next page in the sequence
+   * or the empty string if this is the last page.
+   */
   std::string NextPageToken() const { return raw_page_.next_page_token(); }
+
+  /**
+   * @brief Get the underlying page message.
+   *
+   * This can be useful if the page has custom metadata fields.
```

Comment:
metadata fields? Should this say "fields other than the repeated field or page token?"

Reply:
Will do.